### ISSUE TITLE
Todos on releasing the rewrite into production

### ISSUE THREAD

**Issue by JulianDietzel** [bastislack/highline-freestyle#319]
This is a list of all the things that need to be done once the rewrite reaches a stage where it becomes the main `develop` branch and gets released into production (/ gets shown on the website). This list is continuously updated and meant _not_ for things that need to happen during development leading up to the release but for things that need to be changed / done / updated when the branch or branch name is changed and the app gets published to the website.

If you have the rights to do so feel free to just add items to this list, if not, put your addition in the comments and I will add it.

- [ ] The `.github/workflows/syncsheet.yml` introduced in #315 needs to have its environment variable `TARGET_HEAD_REF` changed from "rewrite" to "develop"
- [ ] Update the GitHub action for publishing the page
- [ ] Update all external packages to release without any known security vulnerabilities
- [ ] At _least_ a week (better more than a week) before the release we should put a banner on the old / current website that there is going to be an update with no backwards compatibility and that no user data can be transferred over to the new app. This way the users can at least prepare somehow (taking notes or screenshots) for the fairly awkward transition.

**Comment by bastislack** [comment]
Just read the point with `no user data can be transferred`. There is really no way? It'll probably be fine, but we need to avoid this in the future.

**Comment by JulianDietzel** [comment]
I agree, We should definetly avoid this in the future. As for if there really is no way:

At the time of writing this I remember being convinced that it would at least take a very large effort to do so. Thinking about it now (for way to little time :D) I think that it might doable to migrate the custom tricks and stick frequencies over but it would take some work. Some considerations:
- In the current (old) state it is possible to modify official tricks. All these modifications won't be possible after the rewrite (for good reasons). Therefore all modified official tricks would need to either be discarded or migrated over into custom tricks (which would cause a duplicate of the official trick and the new custom one (all the sudden you have two Yoda Rolls for no clear reason)). We would also have to check if and how the IDs are treated in the current (old) version when an official trick is changed for the ID to be in the "custom trick range" or not (the current system is that the IDs pace is split into an official and a custom range).
- If we create a migration we need to provide mappings from the IDs of all current official tricks to the corresponding new IDs of those official tricks
- We would need to create a system to detect if the DexieDB browser DB is in the state before or after the rewrite to know when / if to apply the migrations.

All of this (plus maybe some points I have forgotten since then) lead me to believe that the best course of action might be to notify the users ahead of time so that they can somehow save their progression themselfs by writing it down or screenshooting it or something (I know this sucks, I really do), then reset everything with the rewrite and make sure it never happens again. Our new datastructure is more clear and with it, it should be a whole lot easier to prevent such a case in the future.

If you have some time on and hand and feel like creating a migration I would be happy to have a call with you to figure out a solution, it would be awesome. But I personally don't have the capacities to implement this any time soon.
